### ISSUE TITLE
How to build golangci-lint locally?

### ISSUE THREAD

**Issue by iccengan** [golangci/golangci-lint#641]
I have downloaded Golangci-Lint(version 1.17.1)'s source code from the release page and I've made some changes in it.

I am wondering how do I supposed to build various versions if the binaries will be run in different platforms such as darwin-amd64, linux-386 and windows-amd64?

**Comment by jimen0** [comment]
Hi,

Isn't it already described in this section? https://github.com/golangci/golangci-lint/blob/master/README.tmpl.md#local-installation

In regards of cross compilation, should we mention something? Everything needed is there in Go's docs.

If you would want to see more information, please let me know and I'll create a PR.

**Comment by tpounds** [comment]
@iccen8 The instructions linked by @jimen0 are correct if you want to pull a specific version using `go get` in your project. If you wish to contribute/build a local version of `golangci-lint` you can perform the following minimal steps to checkout the source and build the binary.

```shell
git clone https://github.com/golangci/golangci-lint.git
cd golangci-lint
make golangci-lint
```

I'll take a look at the README to see if there is somewhere that makes sense to add it.

**Comment by iccen8** [comment]
I finally ended up with `go build -mod vendor -o golangci-lint ./cmd/golangci-lint` across different platforms and this command satisfied my need. Thank you for your responses.

**Comment by davidpfarrell** [comment]
This ticket showed up in my subscription to CodeTriage but looks like its resolved (since Oct 2019) and ready to close?

**Comment by sayboras** [comment]
this can be closed now
